src/ophyd_async/core/_readable.py

```diff
@@ -210,6 +212,36 @@ def add_readables(
             if isinstance(obj, HasHints):
                 self._has_hints += (obj,)
 
+    @AsyncStatus.wrap
+    async def prepare(self, value: ReadableDeviceConfig) -> None:
+        tasks = []
+        for dtype, signals in value.signals.items():
+            for signal_name, (expected_dtype, val) in signals.items():
+                if hasattr(self, signal_name):
+                    attr = getattr(self, signal_name)
+                    if isinstance(attr, (HintedSignal, ConfigSignal)):
+                        attr = attr.signal
+                    if attr._backend.datatype == expected_dtype:  # noqa: SLF001
+                        if val is not None:
+                            tasks.append(attr.set(val))
+                    else:
+                        raise TypeError(
+                            f"Expected value of type {expected_dtype} for attribute"
+                            f" '{signal_name}',"
+                            f" got {type(attr._backend.datatype)}"  # noqa: SLF001
+                        )
+        await asyncio.gather(*tasks)
```

Comment:
I was motivated to do this because there is no obvious way to configure a device. In StandardReadable there is no method that can use the configure protocol. It looks like `prepare` was intended to be a way to configure a device but there is no obvious way to use it. `prepare` only takes a single argument, intended to be a dataclass, so this is one way a dataclass might be used to write to any of the configuration signals using prepare.

To me, it does not at all seem intuitive to "move" a config signal to some value to set it. As a user or beamline scientist, I suspect this might be perceived as "hacky".

One nice application of these dataclasses is that it provides a data structure for configurations one may use to swap configurations easily within a plan or save/load configurations to and from a file or database.